### ISSUE TITLE
Port Sitemap feature

### ISSUE THREAD

**Issue by mcritchlow** [ucsdlib/horton#55]
### Descriptive summary

We need to recreate the sitemap feature we have in damspas, currently using the following gem [1].

If there is any built-in feature in Hyrax, perhaps we can use that and configure as needed.

If there is not, please port the sitemap feature to Horton.

1. https://github.com/ucsdlib/damspas/blob/develop/Gemfile#L16

A dependency on completing this work is a decision about Capistrano vs Ansistrano and getting the core of that setup. This will be in a separate ticket.

### Rationale

The sitemap feature is needed to support and enhance our SEO.

### Related work

Depends on #56

**Comment by hweng** [comment]
@mcritchlow, I don't find built-in sitemap feature from Hyrax feature list: https://github.com/projecthydra/sufia/wiki/Feature-matrix
And I don't find such config in hyrax project.

**Comment by hweng** [comment]
To create the sitemap,
 it may include static pages:
/about
/help
/contact
/catalog/facet/creator_sim?facet.sort=index
/catalog/facet/topic_sim?facet.sort=index
/catalog/facet/resource_type_sim?facet.sort=index

And all 
object_resources records,
collection records,
subject records,
agent records,
place records,

Is there anything else besides the above would you like to include?

**Comment by mcritchlow** [comment]
@hweng off the top, I think that covers it and, if nothing else, would be an excellent starting point. @lsitu or @VivianChu can you think of anything else?

**Comment by lsitu** [comment]
@mcritchlow Yes, I agree it's a good great starting point. Just not sure whether we need to expose those local authority records (subject, agent, place) in the site map or not.

**Comment by mcritchlow** [comment]
@lsitu yeah, that's a good point. Authorities we general expose only in the context of object and collection records. Perhaps we could leave them out, for now, but have a conversation with @ucsdlib/domm about whether we want to make them more directly visible?

**Comment by hweng** [comment]
@mcritchlow @lsitu Thanks for the comments. I agree to leave out the subject, agent, place for now and discuss it with @ucsdlib/domm.

**Comment by hweng** [comment]
@mcritchlow As we discussed that only public url will be included in the sitemap, which excludes the Open Access, campus and embargoed records url. Please Confirm!

And a question: What is to set the Sitemap.default_host? We haven't deployed it to production yet. But since the ping_search_engines service will not be triggered until the app deploy to production, it can set anything temporarily now.  Do you want me to temporarily set it as "http://library.ucsd.edu/xxx"?

**Comment by mcritchlow** [comment]
Sounds good re: public objects and collections 👍 

And yeah, I suppose that's a reasonable default for the host in other environments since, as you said, it will only be used in production.

**Comment by hweng** [comment]
@mcritchlow Attached is the sitemap that generated from my local. Let me know if it looks good to you and I will make a pull request. 
[sitemap.xml.gz](https://github.com/ucsdlib/horton/files/948250/sitemap.xml.gz)

**Comment by mcritchlow** [comment]
Is this just the default export, or starting point, from the gem? As far as overall structure, everything seems to be in place. I think a PR would be a great idea so the team can look at the config settings. Otherwise, as you noted earlier, we'll need an environment with some useful content to really test this out.

**Comment by hweng** [comment]
@mcritchlow This is not default export from the gem. This is generated from customized configuration for our hyrax project.

**Comment by hweng** [comment]
And from the testings from my local,  only public record urls are pulling out from solr and added to sitemap when I triggered generation task from my local.

**Comment by mcritchlow** [comment]
Ok, thanks for clarifying. I think it was a bit hard to tell looking at the file, with the example.com domain in the URL. That said, sounds like it's definitely worth a PR at this point. Thanks for getting this setup!

**Comment by hweng** [comment]
Yes, the host name is updated to use "http://library.ucsd.edu/hyrax" for now.
